__tests__/chat.test.ts

```diff
@@ -306,18 +310,20 @@ describe('Chat Methods', (): void => {
       const channelAlice = {name: config.bots.bob1.username}
       const channelBob = {name: config.bots.alice1.username}
       const body = 'Dearest Bob, how are you?'
-      let incoming: MsgSummary = null
+      let incoming: MsgSummary | null = null
       const watcher: OnMessage = (message: MsgSummary): void => {
         incoming = message
       }
       bob.chat.watchChannelForNewMessages(channelBob, watcher)
       await timeout(500)
       await alice1.chat.send(channelAlice, {body})
       await timeout(500)
-      if (incoming.content.type !== 'text') {
+
+      const inc = coerceMsgSummary(incoming)
+      if (inc.content?.type !== 'text') {
         throw new Error('got a bad message')
       } else {
-        expect(incoming.content.text.body).toBe(body)
+        expect(inc.content?.text?.body).toBe(body)
```

Comment:
I don't fully understand how jest coordinates tests; how does it handle an `expect` inside a function which may or may not be executed as a callback / emitter elsewhere? Like for example, what if OnMessage is never called? Or called multiple times? Will it keep running the same test?

Reply:
I kind of assumed that inside one of these `it` functions, it would run the fn and if it returned (or resolved) without any problems, then it would pass the test. So if instead you wrote an `it` function taht resolved right away but triggered something else that might later run a function that `expected` something, it would be too late, since the test would be over?

Reply:
Ah I see; good point that if it never gets called this would be able to catch that but moving it into watcher wouldn't. I just assumed if async/await works, it should work from a regular callback as well. But that doesn't matter now since we can't guarantee it happened.